### PR TITLE
GH-34669: [Packaging][Conda] Update arrow feedstock dependencies

### LINKED ISSUE #34669: [Packaging][Conda] Sync with conda-forge feedstock
### Describe the enhancement requested

Regular hyghiene, like in https://github.com/apache/arrow/pull/14832

### Component(s)

C++, Packaging, Python

### PR DESCRIPTION
**Pull request by h-vetinari**
As well as other changes that happened on the feedstock since #14832
* Closes: #34669

### CODE REVIEW

File: dev/tasks/conda-recipes/arrow-cpp/activate.sh
Comment (h-vetinari) [inline dev/tasks/conda-recipes/arrow-cpp/activate.sh:19]: Ah, this won't work unless we carry something like https://github.com/conda-forge/arrow-cpp-feedstock/blob/main/recipe/patches/0001-don-t-bake-non-relocatable-CMAKE_INSTALL_FULL_LIBDIR.patch here... It could be done just within `dev/tasks/conda-recipes`, depends a bit if people want that.

The default install doesn't work with the conda-forge infrastructure because it bakes the wrong PREFIX (the one that's there while the package is being created, rather than installed) into the gdb path. For details see https://github.com/conda-forge/arrow-cpp-feedstock/issues/935

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Review by kou — COMMENTED** [review]
Could you open an issue for this pull request and associate it with this pull request?

File: dev/tasks/conda-recipes/arrow-cpp/patches/0001-don-t-bake-non-relocatable-CMAKE_INSTALL_FULL_LIBDIR.patch
Comment (h-vetinari) [reply dev/tasks/conda-recipes/arrow-cpp/patches/0001-don-t-bake-non-relocatable-CMAKE_INSTALL_FULL_LIBDIR.patch:20]: Done: https://github.com/apache/arrow/issues/34670

**Comment by h-vetinari** [comment]
> Could you open an issue for this pull request and associate it with this pull request?

Done.

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by assignUser** [comment]
Thanks a lot for your work on the arrow feedstocks here and on conda-forge @h-vetinari ! 

Does this sync changes from the conda-forge feedstock into the arrow repo or will these changes also be used by the upstream repo?

**Comment by assignUser** [comment]
Ah great, thanks again. I don't think we have any documentation about the whole process of syncing and maintaining the conda-forge feedstock other than a small checkbox as a post-release task... do you think it would make sense to document the process to make it easier for the rest of the community to aid in this process?

**Comment by h-vetinari** [comment]
> do you think it would make sense to document the process to make it easier for the rest of the community to aid in this process?

I've been piecing things together myself as well, but I do have a bunch of experience from the conda-forge side. I could write something up - could you tell me where that should go?

**Comment by assignUser** [comment]
Oh great! I think that would be best located in the post release task section of the release management guide: https://github.com/apache/arrow/blob/main/docs/source/developers/release.rst

**Comment by h-vetinari** [comment]
> I think that would be best located in the post release task section of the release management guide

Thanks. The release side is usually not an issue (though I can document that part as well); the more important part is the day-to-day development - I guess that should be part of the development guide (or similar resources)?

**Comment by assignUser** [comment]
> day-to-day development

What do you mean by this? How to keep the repos in sync  etc. or more conda-forge day to day maintenance? 
I think the release guide is an adequate place to document dev practices for external repos like conda forge, vcpkg, homebrew etc. vs in the dev guide that is imo about arrow development. But maybe the sections for that stuff in the release guide have grown enough to refactor that into a seperate page?

I don't have hard opinions here and am open for your input!

**Comment by h-vetinari** [comment]
> > day-to-day development
> 
> What do you mean by this? How to keep the repos in sync etc. or more conda-forge day to day maintenance?

Yes. None of the things that need to be done on the arrow-side have anything do with releases. In fact the most burning task IMO would be to tackle the tests we've had to skip in conda-forge, because those represent real failures that user of the conda-forge packaging might run into. For this reason I generally try to avoid these skips, but despite pings on this and opening issues, this hasn't gotten tackled yet; the alternative is that releases in conda-forge are indefinitely blocked on such failing tests, which is generally not a happy place to be in either (especially with how ABI-migrations work in conda-forge).

The release-to-conda-forge thing is really secondary, because it's a competely separate infrastructure, with different tools, norms, authorized users etc. (though every interested arrow contributor could become a feedstock maintainer, of course!). I'll happily document these things, but it's much more important how to keep the CI in arrow up-to-date, know how to fix issues as they arise, and make sure the tests pass.

File: dev/tasks/conda-recipes/arrow-cpp/meta.yaml
Comment (h-vetinari) [inline dev/tasks/conda-recipes/arrow-cpp/meta.yaml:395]: I'm talking about these skips (some more serious than others)

**Comment by assignUser** [comment]
Oh I see, thanks for the details! That would indeed probably make sense somewhere in the Dev guide. Not exactly sure where but maybe @kou or @raulcd have ideas?

**Comment by raulcd** [comment]
> Oh I see, thanks for the details! That would indeed probably make sense somewhere in the Dev guide. Not exactly sure where but maybe @kou or @raulcd have ideas?

The only documentation I know about is on this README: https://github.com/apache/arrow/blob/main/dev/tasks/conda-recipes/README.md
I am ok moving the info to the dev guide. Maybe we could have some info on how we update conda, brew, vcpkg, conan, etcetera but as, from what I understand, those are not part of the Arrow release itself we might want to add a note specifying that those packages might take longer to be available, just to manage expectations of users.

**Comment by kou** [comment]
I think that https://github.com/apache/arrow/blob/main/dev/tasks/conda-recipes/README.md is better than the dev guide. Because we don't need to version the information (how to synchronize our configurations with conda-forge's (upstream's) one). We need to know just the latest information.

For Conan, we have a script to synchronize: https://github.com/apache/arrow/blob/main/ci/conan/merge_upstream.sh

I think that an automated script is more useful than documentation + manual operations.

(This is out-of-scope of this pull request.)

**Comment by amol-** [comment]
@raulcd @assignUser @kou is there something still needed on the PR itself apart rebasing? I saw that the discussion got a bit wider than the specific scope of the PR and thus I was wondering

**Comment by assignUser** [comment]
I think the only issue left really is the documentation which can totally be a follow up. @h-vetinari if you resolve the conflict I think we can merge. The patch for https://github.com/apache/arrow/issues/34670 can also be removed as it was merged IIUC?

**Comment by h-vetinari** [comment]
I've been on leave the last few days and until next week. I'll try to get to it after that.

**Comment by assignUser** [comment]
Ah no problem. of course. But with the freeze date for 12.0.0 coming up on monday it would be nice to get this in before. If you don't mind I can take on the rebasing/conflict resolution for now?

**Comment by h-vetinari** [comment]
Sure go ahead. Seems that some test skips got removed on main (which would be great)?

**Comment by assignUser** [comment]
@github-actions crossbow submit -g conda

**Comment by assignUser** [comment]
@h-vetinari the plasma test skip was removed (as plasma was removed). I am not sure how to integrate the changes from #34670 here, so I leave that as is for now. Testing the rebased commit for now. Worst case we update it in a follow up :shrug:

**Comment by kou** [comment]
> I am not sure how to integrate the changes from #34670 here

Could you add `-DARROW_GDB_INSTALL_DIR=replace_this_section_with_absolute_slashed_path_to_CONDA_PREFIX` to `cmake` command lines?

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by h-vetinari** [comment]
now also synced with https://github.com/conda-forge/r-arrow-feedstock/pull/61

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by h-vetinari** [comment]
Bunch of errors on windows + R seem to have cropped up since 11.0 (as https://github.com/conda-forge/r-arrow-feedstock/pull/61 is passing):
```
arrowExports.cpp:1056:17: error: use of undeclared identifier 'acero'
std::shared_ptr<acero::ExecNode> ExecNode_Scan(const std::shared_ptr<acero::ExecPlan>& plan, const std::shared_ptr<ds::Dataset>& dataset, const std::shared_ptr<compute::Expression>& filter, cpp11::list projection);
                ^
arrowExports.cpp:1056:70: error: use of undeclared identifier 'acero'
std::shared_ptr<acero::ExecNode> ExecNode_Scan(const std::shared_ptr<acero::ExecPlan>& plan, const std::shared_ptr<ds::Dataset>& dataset, const std::shared_ptr<compute::Expression>& filter, cpp11::list projection);
                                                                     ^
arrowExports.cpp:1059:40: error: use of undeclared identifier 'acero'
        arrow::r::Input<const std::shared_ptr<acero::ExecPlan>&>::type plan(plan_sexp);    
```

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by kou** [comment]
Could you rebase on the current main for R errors?

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by h-vetinari** [comment]
Still fails with the acero errors for R on windows...

**Comment by kou** [comment]
We can ignore the error for R on Windows in this pull request because it's happen in the last nightly CI too:
https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=47053&view=logs&j=4c86bc1b-1091-5192-4404-c74dfaad23e7&t=ec99a26b-0264-5e86-36fb-9cfd0ca0f9f3&l=1184

Are other timeout errors related to this change?
* https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=47060&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=d9b15392-e4ce-5e4c-0c8c-b69645229181
* https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=47060&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=d9b15392-e4ce-5e4c-0c8c-b69645229181
* https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=47061&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=d9b15392-e4ce-5e4c-0c8c-b69645229181

**Comment by h-vetinari** [comment]
> Are other timeout errors related to this change?

Possibly. I'd just switch off running the test suite for the builds running in emulation TBH.

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by h-vetinari** [comment]
This is now passing except R on windows. CC @kou @assignUser

**Review by kou — APPROVED** [review]
+1

Thanks!

**Comment by assignUser** [comment]
@h-vetinari Thanks for pushing this through!

**Pull request merged**
